### PR TITLE
KEYCLOAK-16935 Fix liquibase to work with MySQL 8.0.23+

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/conn/CustomChangeLogHistoryService.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                    String id = rs.get("ID").toString();
+                    String md5sum = rs.get("MD5SUM") == null || getDatabaseChecksumsCompatible() ? null : rs.get("MD5SUM").toString();
+                    String description = rs.get("DESCRIPTION") == null ? null : rs.get("DESCRIPTION").toString();
+                    String comments = rs.get("COMMENTS") == null ? null : rs.get("COMMENTS").toString();
+                    Object tmpDateExecuted = rs.get("DATEEXECUTED");
+                    Date dateExecuted = null;
+                    if (tmpDateExecuted instanceof Date) {
+                        dateExecuted = (Date) tmpDateExecuted;
+                    } else if (tmpDateExecuted instanceof LocalDateTime) {
+                        dateExecuted = Date.from(((LocalDateTime) tmpDateExecuted).atZone(ZoneId.systemDefault()).toInstant());
```
Comment (hmlnarik) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/conn/CustomChangeLogHistoryService.java:74]: This is the only difference as compared to `StandardChangeLogHistoryService.getRanChangeSets`, and is based on the original fix in liquibase: https://github.com/liquibase/liquibase/commit/90c523562fc6f502d439ba1670bb0793a6467a6c

**Comment by hmlnarik** [comment]
https://master-jenkins/job/universal-test-pipeline-server/272/

**Review by mposolda — APPROVED** [review]
@hmlnarik Thanks for the fix! I am approving (I consider that update liquibase to the proper version with the fix is not an option and hence we don't have other option than fix on our side and add custom changeLogHistoryService class)

**Comment by hmlnarik** [comment]
Pipelines have not run yet due to infrastructure hiccups.

**Comment by hmlnarik** [comment]
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2659

**Pull request merged**